pkg/scale/bitvec.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package scale
+
+const (
+	// maxLen equivalent of `ARCH32BIT_BITSLICE_MAX_BITS` in parity-scale-codec
+	maxLen = 268435455
+	// byteSize is the number of bits in a byte
+	byteSize = 8
+)
+
+// BitVec represents rust's `bitvec::BitVec` in SCALE
+// It is encoded as a compact u32 representing the number of bits in the vector
+// followed by the actual bits, rounded up to the nearest byte
+type BitVec interface {
+	// Bits returns the bits in the BitVec
+	Bits() []uint8
```

Comment:
We're keeping it basic for now since this is being worked on for ParachainHost runtime calls. I've updated the PR and the issue.

Reply:
For this PR, I've found it easy to work with `[]uint`. If it makes sense to change it bool in the future, we could do that.

Reply:
I take a look at how substrate implements `BitVec` and they use a crate called `bitvec` ([scale-codec dependency](https://github.com/paritytech/parity-scale-codec/blob/aaea65ab7b7d0cded3e37f42a819ccedb5db645f/Cargo.toml#LL16C4-L16C4), [create repo](https://github.com/ferrilab/bitvec)), and in the README of this create they explain:

> bitvec provides a foundational API for bitfields in Rust. It specializes standard-library data structures (slices, arrays, and vectors of bool) to use one-bit-per-bool storage, similar to [std::bitset<N>](https://en.cppreference.com/w/cpp/utility/bitset) and [std::vector<bool>](https://en.cppreference.com/w/cpp/container/vector_bool) in C++.

So, I agree with @kishansagathiya that it would be more compatible to use a slice of `bool` as it is used in `parity-scale-codec`.

Reply:
Thanks for looking that up @EclesioMeloJunior . I'll see if I can do it in this PR itself!